model/infinispan/src/main/java/org/keycloak/marshalling/Marshalling.java

```diff
@@ -1,25 +1,155 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.keycloak.marshalling;
 
 import org.infinispan.configuration.global.GlobalConfigurationBuilder;
-import org.infinispan.jboss.marshalling.core.JBossUserMarshaller;
-import org.keycloak.models.sessions.infinispan.changes.ReplaceFunction;
 
-@SuppressWarnings("removal")
+/**
+ * Ids of the protostream type.
+ * <p>
+ * Read careful the following warning to ensure compatibility when updating schemas.
+ * <p>
+ * WARNING! IDs lower or equal than 65535 are reserved for internal Inifinispan classes and cannot be used.
+ * WARNING! ID defined in this class must be unique. If one type is removed, its ID must not be reused. You have been
+ * warned! The ID identifies the message, and it is stored and used to save space.
+ * WARNING! The field IDs cannot be reused as well for the same reason.
+ * WARNING! Primitive types cannot be null in proto3 syntax (Integer, String). Take that in consideration.
+ * <p>
+ * Be Aware of the following default in Proto3 syntax!
+ * For strings, the default value is the empty string.
+ * For bytes, the default value is empty bytes.
+ * For bools, the default value is false.
+ * For numeric types, the default value is zero.
+ * For enums, the default value is the first defined enum value, which must be 0.
+ * For message fields, the field is not set. (null)
+ * <p>
+ * Docs: <a href="https://protobuf.dev/programming-guides/proto3/">Language Guide (proto 3)</a>
+ */
 public final class Marshalling {
 
     private Marshalling() {
     }
 
-    // Note: Min ID is 2500
-    public static final Integer REPLACE_FUNCTION_ID = 2500;
+    // Model
```

Comment:
We can add a test to ensure that all of the ID values in this class are unique, in the Infinispan repo we do:

```java

   @Test
   public void testIdUniqueness() throws Exception {
      Class clazz = ProtoStreamTypeIds.class;
      Field[] fields = clazz.getFields();
      Set<Integer> messageIds = new HashSet<>();
      Set<Integer> lowerBounds = new HashSet<>();
      for (Field f : fields) {
         if (f.getName().endsWith("_LOWER_BOUND"))
            assertTrue(f.getName(), lowerBounds.add(f.getInt(clazz)));
         else
            assertTrue(f.getName(), messageIds.add(f.getInt(clazz)));
      }
      assertTrue(!messageIds.isEmpty());
      assertTrue(!lowerBounds.isEmpty());
      assertEquals(fields.length - lowerBounds.size(), messageIds.size());
      assertEquals(fields.length - messageIds.size(), lowerBounds.size());
   }
```

In this case the test will be simpler as we don't define LOWER_BOUND fields.

Reply:
It is not required. If you have a duplicate ID, the build will fail with 

`org.infinispan.protostream.DescriptorParserException: Duplicate type id 65541 for type keycloak.LockEntry. Already used by keycloak.LockEntryPredicate`

Reply:
I'm obviously behind on the times. We can probably remove that test from Infinispan now then :slightly_smiling_face: